### PR TITLE
Added custom slugify function to allow any name for Repo, Vocab, Term

### LINKED ISSUE #773: User can create empty slug using a character like '('
If something which uses slugs, like a repository, vocabulary, or term, uses invalid characters for a name or label, the newly generated slug will be an empty string. This can cause many problems and we should change our slug generation methods to add some prefix to the slug to guard against this case.

I think this is a severe bug but not enough to delay the next release

### PR DESCRIPTION
**Pull request by giocalitri**
With this PR it will be possible to use names like "%^*&^" for Repo, Vocab and Term

Fixes #773

### CODE REVIEW

File: rest/util.py [hunk 1]
```diff
@@ -60,3 +61,19 @@ def dispatch(self, request, *args, **kwargs):
             CheckValidMemberParamMixin,
             self
         ).dispatch(request, *args, **kwargs)
+
+
+def default_slugify(label, default_name, exists_func):
```
Comment (noisecapella) [inline rest/util.py:66]: I'm not sure on the usefulness of `default_name` here. Why not just use 'slug' as a default term no matter where the slug is assigned? No strong feelings however

Comment (giocalitri) [inline rest/util.py:66]: the reason was because it looked weid to me to have a term and a repository have the same base name.

File: rest/util.py [hunk 2]
```diff
@@ -60,3 +61,19 @@ def dispatch(self, request, *args, **kwargs):
...
+    """
+    Function that extends the Django `slugify` to add a default string in case
```
Comment (noisecapella) [inline rest/util.py:68]: The arguments and return value need documentation here

Comment (giocalitri) [inline rest/util.py:68]: ok

**Comment by noisecapella** [comment]
This needs tests for repository slugs, other than that everything looks fine

**Comment by giocalitri** [comment]
Comments addressed

**Comment by noisecapella** [comment]
I forgot to mention this earlier but we should also have migrations to rename existing empty slugs

**Comment by giocalitri** [comment]
ok. I'm starting to work on it

**Comment by giocalitri** [comment]
Migrations added

**Comment by justinabrahms** [comment]
Looks pretty good to me, though I don't have much context on the overarching code base. I agree with @noisecapella that the API is a little awkward to pass a default name in, but that it's also not a huge deal. 

Good stuff! :smile_cat:

**Comment by noisecapella** [comment]
Migrations work for me, feel free to merge. You'll have to repush due to Travis intermittent failures

**Comment by giocalitri** [comment]
@noisecapella you do not need to re-push to restart travis: if you login in travis (with the github credentials) you can see a button to restart the job.

**Comment by giocalitri** [comment]
thanks, @justinabrahms

**Comment by giocalitri** [comment]
Tests passed: merging

**Pull request merged**